cmd/agent/app/options/options.go

```diff
@@ -117,6 +119,7 @@ func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
 	flags.StringVar(&o.AgentIdentifiers, "agent-identifiers", o.AgentIdentifiers, "Identifiers of the agent that will be used by the server when choosing agent. N.B. the list of identifiers must be in URL encoded format. e.g.,host=localhost&host=node1.mydomain.com&cidr=127.0.0.1/16&ipv4=1.2.3.4&ipv4=5.6.7.8&ipv6=:::::&default-route=true")
 	flags.BoolVar(&o.WarnOnChannelLimit, "warn-on-channel-limit", o.WarnOnChannelLimit, "Turns on a warning if the system is going to push to a full channel. The check involves an unsafe read.")
 	flags.BoolVar(&o.SyncForever, "sync-forever", o.SyncForever, "If true, the agent continues syncing, in order to support server count changes.")
+	flags.BoolVar(&o.EnableAgentServerReadiness, "agent-server-readiness", o.EnableAgentServerReadiness, "If true, the agent readiness probe checks if its connected to atleast one server.")
```

Comment:
> flags.BoolVar(&o.EnableAgentServerReadiness, "agent-server-readiness",

Did you add this flag based on a misunderstanding with @cheftako comments (about changing liveness behavior)? I don't remember saying that we need a flag to enable the new readiness behavior (I am ok with not flag guarding the readiness change).

If we keep the flag: since this is the agent binary, "agent" in name seems somewhat redundant. Maybe "--enable-connected-readiness" or similar?

Reply:
Yes, I think there was a confusion on that. I too don't think a flag is necessary.
I'm happy to remove the commit related to addition of the flag.

Reply:
removed the commit.